state/application_test.go

```diff
@@ -1928,6 +1931,69 @@ func (s *ApplicationSuite) TestAddCAASUnit(c *gc.C) {
 	})
 }
 
+func (s *ApplicationSuite) TestAgentTools(c *gc.C) {
+	st := s.Factory.MakeModel(c, &factory.ModelParams{
+		Name: "caas-model",
+		Type: state.ModelTypeCAAS, CloudRegion: "<none>",
+		StorageProviderRegistry: factory.NilStorageProviderRegistry{}})
+	defer st.Close()
+	f := factory.NewFactory(st)
+	application := f.MakeApplication(c, &factory.ApplicationParams{
+		CharmConfig: map[string]interface{}{
+			"foo": "bar",
+		},
+		ApplicationConfig: map[string]interface{}{
+			"app foo": "app bar",
+		},
+		ApplicationConfigFields: environschema.Fields{
+			"app foo": environschema.Attr{Type: environschema.Tstring}},
+	})
+	agentTools := version.Binary{
+		Number: jujuversion.Current,
+		Arch:   arch.HostArch(),
+		Series: application.Series(),
+	}
+
+	tools, err := application.AgentTools()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(tools.Version, gc.Equals, agentTools)
+}
+
+func (s *ApplicationSuite) TestSetAgentVersion(c *gc.C) {
+	st := s.Factory.MakeModel(c, &factory.ModelParams{
+		Name: "caas-model",
+		Type: state.ModelTypeCAAS, CloudRegion: "<none>",
+		StorageProviderRegistry: factory.NilStorageProviderRegistry{}})
+	defer st.Close()
+	f := factory.NewFactory(st)
+	application := f.MakeApplication(c, &factory.ApplicationParams{
+		CharmConfig: map[string]interface{}{
+			"foo": "bar",
+		},
+		ApplicationConfig: map[string]interface{}{
+			"app foo": "app bar",
+		},
+		ApplicationConfigFields: environschema.Fields{
+			"app foo": environschema.Attr{Type: environschema.Tstring}},
+	})
+	agentTools := version.Binary{
+		Number: jujuversion.Current,
```

Comment:
version.MustParse("2.99.0")

(or some other fake version number)
We want it to be different to version.Current